core/api-keys/src/identity/mod.rs

```diff
@@ -105,4 +105,39 @@ impl Identities {
             Err(IdentityError::NoActiveKeyFound)
         }
     }
+
+    pub async fn list_keys_for_subject(
+        &self,
+        subject_id: &str,
+    ) -> Result<Vec<IdentityApiKey>, IdentityError> {
+        let identity = sqlx::query!(
+            r#"SELECT id FROM identities WHERE subject_id = $1"#,
+            subject_id,
+        )
+        .fetch_optional(&self.pool)
+        .await?
+        .ok_or(IdentityError::NoIdentityForSubject)?;
```

Comment:
Should we create the identity here and return an empty array instead?

Reply:
not here but in app layer we should return an empty list

Reply:
Solved with join